datasets/us/dc/med_exclusions/crawler.py

```diff
@@ -0,0 +1,85 @@
+from zavod import Context, helpers as h
+
+
+def log_messy_names(context, name_value):
+    if any(term in name_value for term in ("aka", "a/k/a")):
+        context.log.warning("Name needs to be cleaned up:", name=name_value)
+
+
+def lookup_position(context, principal):
+    position = None
+    if "President" in principal:
+        result = context.lookup("director", principal)
+        if result and result.names:
+            principal = result.names[0].get("principal")
+            position = result.names[0].get("position")
+        if not position:
+            context.log.warning(
+                "Could not find position for director", director=principal
+            )
+    return principal, position
+
+
+def emit_directorship(context, entity_id, principal, position):
+    director = context.make("Person")
+    director.id = context.make_id(principal)
+    director.add("name", principal)
+    log_messy_names(context, director.get("name")[0])
+    director.add("position", position)
+    context.emit(director)
+
+    dir = context.make("Directorship")
+    dir.id = context.make_id(director.id, "director of", entity_id)
+    dir.add("organization", entity_id)
+    dir.add("director", director.id)
+    context.emit(dir)
+
+
+def crawl_row(context, row):
+    name = row.pop("name_of_individual", None)
+    company_name = row.pop("name_of_company", None)
+    schema = "Person" if name else "Company"
+
+    entity = context.make(schema)
+    entity.id = context.make_id(name, company_name)
+    if entity.schema.is_a("Person"):
+        entity.add("name", name)
+        log_messy_names(context, entity.get("name")[0])
+    else:
+        entity.add("name", company_name)
```

Comment:
It feels like the code isn't really reflecting how there's only one name and it's either an individual name, or a company name. It's a bit of a nitpick but I think the code would be a bit clearer if we end up assigning to `name_raw`.

Part of what I think makes the current approach a bit weird how we're making an id where either of those are expected to have a value. It might actually be good to add an assertion that you end up with _some_ value in name_raw.

Let's use name_raw and address for id just in case there are two different John Smiths getting banned.